source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/RatingInputHandler.kt

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.adaptivecards.renderer.inputhandler
+
+import android.view.accessibility.AccessibilityEvent
+import io.adaptivecards.objectmodel.BaseInputElement
+import io.adaptivecards.objectmodel.RatingInput
+import io.adaptivecards.renderer.Util
+import io.adaptivecards.renderer.layout.RatingStarInputView
+import io.adaptivecards.renderer.layout.RatingStarInputViewListener
+
+/**
+ * Input handler for Rating input
+ **/
+class RatingInputHandler(
+    baseInputElement: BaseInputElement
+): BaseInputHandler(baseInputElement) {
+    override fun getInput() = (m_view as RatingStarInputView).rating.toString()
+
+    override fun setInput(input: String) {
+        try {
+            (m_view as RatingStarInputView).handleRatingChange(input.toDouble())
+        } catch(e: NumberFormatException) {
+            return
+        }
+    }
+
+    override fun setFocusToView() {
+        val focusView = (m_view as RatingStarInputView).getChildAt(0)
+        Util.forceFocus(focusView)
+        focusView.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED)
+    }
+
+    override fun isValid(showError: Boolean): Boolean {
+        var isValid = true
+        if (m_baseInputElement.GetIsRequired()) {
+            isValid = try {
+                input.toDouble() > 0
+            } catch(e: NumberFormatException) {
+                false
+            }
+        }
+        isValid = isValid && isValidOnSpecifics(input)
+        if (showError) {
+            showValidationErrors(isValid)
+        }
+        return isValid
+    }
+
+    override fun resetValue() {
+        val ratingInput = Util.castTo(m_baseInputElement, RatingInput::class.java)
+        input = ratingInput.GetValue().toString()
+    }
+
+    override fun registerInputObserver() {
+        (m_view as RatingStarInputView).setRatingStarInputViewListener(object: RatingStarInputViewListener {
+            override fun onRatingChanged() {
+                notifyAllInputWatchers()
+            }
+        })
```

Comment:
Can you please add `addValueChangedActionInputWatcher();` at the end of `registerInputObserver()`. ref -  other InputHandler.

Reply:
I didn't add this because as per my last discussion, we were not supporting value changed action for rating element.

Adding now.

Reply:
Added now.